src/main/groovy/io/seqera/wave/service/CredentialServiceImpl.groovy

```diff
@@ -92,7 +92,7 @@ class CredentialServiceImpl implements CredentialsService {
         // log for debugging purposes
         log.debug "Credentials matching criteria registryName=$registryName; userId=$identity.userId; workspaceId=$identity.workspaceId; endpoint=$identity.towerEndpoint => $creds"
```

Comment:
should we log workflowId too?

Reply:
The logic of this change is to use two caches, one short (when workflowId is absent) and the long (when workflowId) is provided.

The cache key should determined by the `cacheKey` parameter which is computed ad-hoc c   

https://github.com/seqeralabs/wave/blob/9122a302cf63d11fb086916b4d59fe239385f363/src/main/groovy/io/seqera/wave/tower/client/TowerClient.groovy#L52-L60

In the case of user-info is made by the URI and the auth key 

https://github.com/seqeralabs/wave/blob/9122a302cf63d11fb086916b4d59fe239385f363/src/main/groovy/io/seqera/wave/tower/client/TowerClient.groovy#L64-L64

In principle the `cacheKey` should be identical and therefore there should be only one request. Either the cacheKey is not unique or micronaut cache is failing somewhere. 

I'd suggest making a local test and checking with multiple concurrent requests

Reply:
ok sure
I will work on it

Reply:
I tried adding custom keyGenerator to make sure that its only using the cacheKey as key, but still not resolved
https://github.com/micronaut-projects/micronaut-cache/issues/71#issuecomment-583538488

digging mre into it